src/client/main.cpp

```diff
@@ -17,59 +18,52 @@
 
 #include "frida/frida-gum.h"
 
-#define FB_ID "mxcfb"
-
-#ifndef _GNU_SOURCE
-#define _GNU_SOURCE
-#endif
-
-int msg_q_id = 0x2257c;
+constexpr auto msg_q_id = 0x2257c;
 swtfb::ipc::Queue MSGQ(msg_q_id);
 
-uint16_t *SHARED_BUF;
+uint16_t *SHARED_BUF = nullptr;
 
-const int BYTES_PER_PIXEL = sizeof(uint16_t);
+constexpr auto BYTES_PER_PIXEL = sizeof(*SHARED_BUF);
 
 bool IN_XOCHITL = false;
+bool ON_RM2 = false;
 
 extern "C" {
-struct stat;
 
-// QImage(width, height, format)
-static void (*qImageCtor)(void *that, int x, int y, int f) = 0;
-// QImage(uchar*, width, height, bytesperline, format)
-static void (*qImageCtorWithBuffer)(void *that, uint8_t *, int32_t x, int32_t y,
-                                    int32_t bytes, int format, void (*)(void *),
-                                    void *) = 0;
-
-static void _libhook_init() __attribute__((constructor, visibility("default")));
-static void _libhook_init() {
+__attribute__((constructor))
+void init() {
   std::ios_base::Init i;
 
-  auto VERSION = "0.1";
+  constexpr auto VERSION = "0.1";
 
   setenv("RM2FB_SHIM", VERSION, true);
 
-  if (getenv("RM2FB_ACTIVE") != "") {
+  if (getenv("RM2FB_ACTIVE") != nullptr) {
     setenv("RM2FB_NESTED", "1", true);
   } else {
     setenv("RM2FB_ACTIVE", "1", true);
   }
 
```

Comment:
these should only be set when on RM2. applications that work on rm1/rm2 may use RM2FB_SHIM to detect that they are inside the shim (among other ways)